File: backend/studentattendanceapp/src/main/java/com/example/appbinus/dto/StudentProfileDTO.java
package com.example.appbinus.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class StudentProfileDTO
{
    private String studentNim;
    private String studentName;
    private String studentEmail;
    private String studentTelephone;
    private String studentAddress;
    private String studentMajor;

}

File: Frontend/lib/models/student_profile_dto.dart
class StudentProfileDTO {
  final String studentNim;
  final String studentName;
  final String studentEmail;
  final String? studentTelephone;
  final String? studentAddress;
  final String? studentMajor;

  StudentProfileDTO({
    required this.studentNim,
    required this.studentName,
    required this.studentEmail,
    this.studentTelephone,
    this.studentAddress,
    this.studentMajor,
  });

  factory StudentProfileDTO.fromJson(Map<String, dynamic> json) {
    return StudentProfileDTO(
      studentNim: json['studentNim'] ?? '',
      studentName: json['studentName'] ?? '',
      studentEmail: json['studentEmail'] ?? '',
      studentTelephone: json['studentTelephone'],
      studentAddress: json['studentAddress'],
      studentMajor: json['studentMajor'],
    );
  }
}